.github/workflows/release.yml

```diff
@@ -13,51 +13,69 @@ on:
     runs-on: ubuntu-latest
     steps:
-      - name: Clone source code
+      -
+        name: "Checkout  source code"
         uses: actions/checkout@v1
         with:
           fetch-depth: 1
-          ref: release
-
-      - name: Login to Quay.io
-        uses: azure/docker-login@v1
+      - 
+        name: Prepare
+        id: prep
+        run: |
+          SHORT_SHA1=$(git rev-parse --short HEAD)
+          echo ::set-output name=short_sha1::${SHORT_SHA1}
+          VERSION=$(head -n 1 VERSION)
+          echo ::set-output name=version::${VERSION}
+          IMAGE=che-devfile-registry
+          echo ::set-output name=image::${IMAGE}
+          PLATFORMS=$(cat PLATFORMS)
+          echo ::set-output name=platforms::${PLATFORMS}
+      -
+        name: "Set up QEMU"
+        uses: docker/setup-qemu-action@v1
+      -
+        name: "Set up Docker Buildx"
+        uses: docker/setup-buildx-action@v1
+      -
+        name: "Docker quay.io Login"
+        uses: docker/login-action@v1
         with:
-          login-server: quay.io
+          registry: quay.io
           username: ${{ secrets.QUAY_USERNAME }}
           password: ${{ secrets.QUAY_PASSWORD }}
-
-      - name: Login to Docker Hub
-        uses: azure/docker-login@v1
+      - 
+        name: Login to docker.io
+        uses: docker/login-action@v1
         with:
+          registry: docker.io
           username: ${{ secrets.DOCKERHUB_USERNAME }}
           password: ${{ secrets.DOCKERHUB_PASSWORD }}
-
-      - name: Build and push patched base images and happy path image
+          
+      - 
+        name: Build and push patched base images and happy path image
         run: |
-          export TAG=$(head -n 1 VERSION)
           /bin/bash arbitrary-users-patch/happy-path/build_happy_path_image.sh --push
           /bin/bash arbitrary-users-patch/build_images.sh --push
-
-      - name: Build and push images
-        run: |
-          SHORT_SHA1=$(git rev-parse --short HEAD)
-          VERSION=$(head -n 1 VERSION)
-          IMAGE=che-devfile-registry
-          DOCKERFILE_PATH=./build/dockerfiles/Dockerfile
-          docker build -t ${IMAGE} -f ${DOCKERFILE_PATH} --build-arg PATCHED_IMAGES_TAG=${VERSION} --target registry .
-          docker tag ${IMAGE} quay.io/eclipse/${IMAGE}:${SHORT_SHA1}
-          docker push quay.io/eclipse/${IMAGE}:${SHORT_SHA1}
-          docker tag ${IMAGE} quay.io/eclipse/${IMAGE}:${VERSION}
-          docker push quay.io/eclipse/${IMAGE}:${VERSION}
+      -
+        name: "Build and push"
+        uses: docker/build-push-action@v2
+        with:
+          context: .
+          file: ./build/dockerfiles/Dockerfile
+          platforms: ${{  steps.prep.outputs.platforms }}
+          tags:  quay.io/eclipse/${{ steps.prep.outputs.image }}:${{ steps.prep.outputs.version }},quay.io/eclipse/${{ steps.prep.outputs.image }}:${{ steps.prep.outputs.short_sha1 }},quay.io/eclipse/${{ steps.prep.outputs.image }}:${{  github.event.inputs.version }}
```

Comment:
you're pushing the same image w/ 3 tags here:
*           quay.io/eclipse/${{ steps.prep.outputs.image }}:${{ steps.prep.outputs.version }},
*           quay.io/eclipse/${{ steps.prep.outputs.image }}:${{  github.event.inputs.version }}
*           quay.io/eclipse/${{ steps.prep.outputs.image }}:${{ steps.prep.outputs.short_sha1 }},

should it be only two pushes for prep.outputs.version and short_sha1? Not sure why we would want the submitted version 7.y.z AND version from VERSION file AND the short sha from the tip of the checked out branch. 

@mkuznyetsov WDYT?

Reply:
@nickboldt 

> should it be only two pushes for prep.outputs.version and short_sha1?
Yes
```
prep.outputs.version-->7.27.0-SNAPSHOT
event.inputs.version-->7.27.0 
```
I was trying to release proper version, so that `event.inputs.version` I have added. Please let me know should I remove `event.inputs.version`?

I will remove spaces from `${{ github.event.inputs.version }}`

Reply:
we don't release -SNAPSHOT versions to quay.

https://quay.io/repository/eclipse/che-devfile-registry?tab=tags

Just nightly + sha tag and release (7.yy.z) + sha tag.